lib/puppet/type/cisco_vxlan_vtep_vni.rb

```diff
@@ -0,0 +1,157 @@
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_vxlan_vtep_vni) do
+  @doc = "Manages a Cisco Virtual Tunnel Endpoint (VTEP) to Virtual Network
+  Identifier (VNI) mapping.
+
+  cisco_vxlan_vtep_vni {'<title>':
+    ..attributes..
+  }
+
+  <title> is the title of vxlan_vtep_vni resource.
+
+  Example:
+    cisco_vxlan_vtep_vni {'nve1 10000':
+      ensure      		=> present,
+      assoc_vrf			=> false,
+      ingress_replication       => 'static',
+      multicast_group		=> '224.1.1.1 224.1.1.200',
+      peer_ips			=> ['1.1.1.1', '2.2.2.2', '3.3.3.3'],
+      suppress-arp		=> true,
+    }"
+
+  ###################
+  # Resource Naming #
+  ###################
+
+  # Parse out the title to fill in the attributes in these
+  # patterns. These attributes can be overwritten later.
+  def self.title_patterns
+    identity = ->(x) { x }
+    patterns = []
+
+    # Below pattern matches both parts of the full composite name.
+    patterns << [
+      /^(\S+)\s+(\d+)$/,
+      [
+        [:interface, identity],
+        [:vni, identity],
+      ],
+    ]
+    patterns
+  end
+
+  newparam(:interface, namevar: :true) do
+    desc 'Name of the nve interface on the network element.
+          Valid values are string.'
+
+    validate do |value|
+      fail('Interface name must be a string') unless value.is_a? String
+    end
+
+    munge(&:downcase)
+  end
+
+  newparam(:vni, namevar: true) do
+    desc 'ID of the Virtual Network Identifier. Valid values are integer.'
+  end
+
+  newparam(:assoc_vrf, namevar: true) do
+    desc 'Associate vrf with the vni. Valid values are true or false.'
+    defaultto(:false)
+    newvalues(:true, :false)
+  end
+
+  # param id
+
+  ##############
+  # Attributes #
+  ##############
+
+  ensurable
+
+  # Overwrites the name method which by default returns only self[:name].
+  def name
+    "#{self[:interface]} #{self[:vni]}"
+  end
+
+  # Only needed to satisfy name parameter.
+  newparam(:name) do
+  end
+
+  newproperty(:ingress_replication) do
+    desc "Specify mechanism for host reachability advertisement. Valid values
+          are 'bgp', 'static', or 'default'"
+
+    newvalues(:bgp, :static, :default)
+  end
+
+  newproperty(:multicast_group) do
+    desc "The multicast group (range) of the VNI. Valid values are string,
+          keyword 'default'"
+    munge do |multicast_group|
+      multicast_group = :default if multicast_group == 'default'
+      fail 'multicast_group is not a string.' unless
+        multicast_group == :default || multicast_group.is_a?(String)
+      multicast_group
+    end
+  end # property name
+
+  newproperty(:peer_ips, array_matching: :all) do
+    desc 'A list of static ip addresses for ingress-replication static' \
+         "valid values are an array of strings and keyword 'default'."
+
+    validate do |value|
+      fail "peer_ips #{value} must be string" unless
+        value.kind_of?(String) || value == :default
```

Comment:
Given that your are looping through each of these peers, you can use the following method to validate that they are valid ip's instead of just strings.

https://github.com/cisco/cisco-network-puppet-module/blob/release_1.2.0/lib/puppet/type/cisco_bgp_neighbor_af.rb#L234